### PR TITLE
Refactor Pause state (1/2)

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #4556 

### Summary of Changes

- cleanup frame mapping for original locations
- extract togglePrettyPrint
- polish togglePrettyPrint UX
- removed pause.pause and added a stopgap

### CODE REVIEW

File: src/components/SecondaryPanes/Breakpoints.js [hunk 1]
```diff
@@ -52,12 +52,12 @@ type Props = {
 };
 
 function isCurrentlyPausedAtBreakpoint(pause, breakpoint) {
-  if (!pause || pause.isInterrupted) {
+  if (!pause || pause.why.type == "isInterrupted") {
```
Comment (bomsy) [inline src/components/SecondaryPanes/Breakpoints.js]: should it be`pause.why.type === "interrupted"`?

File: src/reducers/pause.js [hunk 2]
```diff
@@ -254,6 +260,11 @@ export function getSelectedFrameId(state: OuterState) {
   return state.pause.selectedFrameId;
 }
 
+export function getTopmostFrame(state: OuterState) {
```
Comment (bomsy) [inline src/reducers/pause.js]: naming nit: How about `getTopFrame`

File: src/actions/sources/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export { createPrettySource } from "./createPrettySource";
+export { loadSourceText } from "./loadSourceText";
+export { togglePrettyPrint } from "./togglePrettyPrint";
```
Comment (bomsy) [inline src/actions/sources/index.js]: Ws seem to be having lots of actions files that seem to have just one action. Wondering if we can group related actions
e.g `createPrettySource` and `togglePrettyPrint` can go into one file `prettyPrint.js`

```js
export { createPrettySource, togglePrettyPrint } from "./prettyPrint";
```

Comment (jasonLaster) [reply src/actions/sources/index.js]: good idea. now that we have extracted `togglePrettyPrint` we can consolidate the two!

File: src/reducers/pause.js (hunk 2 above)
Comment (jasonLaster) [reply src/reducers/pause.js]: Oh nice! I started with that :)

File: src/actions/sources.js [hunk 4]
```diff
@@ -258,12 +256,18 @@ export function selectLocation(location: Location, tabIndex: string = "") {
...
-    dispatch(setOutOfScopeLocations());
-    const src = getSource(getState(), location.sourceId);
-    const { autoPrettyPrint } = prefs;
-    if (autoPrettyPrint && shouldPrettyPrint(src) && isMinified(src)) {
-      await dispatch(togglePrettyPrint(src.get("id")));
+    const selectedSource = getSelectedSource(getState());
+    if (
+      prefs.autoPrettyPrint &&
+      !getPrettySource(getState(), source.get("id")) &&
+      shouldPrettyPrint(selectedSource) &&
```
Comment (bomsy) [inline src/actions/sources.js]: Is `selectedSource` different from `source`? Both are used here.

File: src/utils/pause.js [hunk 5]
```diff
@@ -3,29 +3,10 @@
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
 // @flow
-import type { Pause, Frame } from "../types";
+import type { Pause } from "../types";
 import { get } from "lodash";
```
Comment (bomsy) [inline src/utils/pause.js]: No longer used

**Comment by jasonLaster** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=48eecd7ea9e8eccc60e1f30eeae8b6faa24a688b

**Pull request merged**